development/cpp/unit_testing.rst

```diff
@@ -0,0 +1,288 @@
+.. _doc_unit_testing:
+
+Unit testing
+============
+
+Godot Engine allows to write unit tests directly in C++. The engine integrates
+the `doctest <https://github.com/onqtam/doctest>`_ unit testing framework which
+gives ability to write test suites and test cases next to production code, but
+since the tests in Godot go through a different ``main`` entry point, the tests
+reside in a dedicated ``tests/`` directory instead, which is located at the root
+of the engine source code.
+
+Platform and target support
+---------------------------
+
+C++ unit tests can be run on Linux, MacOS, and Windows operating systems.
+
+Tests can only be run with editor ``tools`` enabled, which means that export
+templates cannot be tested currently.
+
+Running tests
+-------------
+
+Before tests can be actually run, the engine must be compiled with the ``tests``
+build option enabled (and any other build option you typically use), as the
+tests are not compiled as part of the engine by default:
+
+.. code-block:: shell
+
+    scons tests=yes
+
+Once the build is done, run the tests with a ``--test`` command-line option:
+
+.. code-block:: shell
+
+    ./bin/<godot_binary> --test
+
+The test run can be configured with the various doctest-specific command-line
+options. To retrieve the full list of supported options, run the ``--test``
+command with the ``--help`` option:
+
+.. code-block:: shell
+
+    ./bin/<godot_binary> --test --help
+
+Any other options and arguments after the ``--test`` command are treated as
+arguments for doctest.
+
+.. note::
+
+    Tests are compiled automatically if you happen to use ``dev=yes`` option,
+    which is recommended to supply if you plan on contributing to the engine
+    development, as some compilation warnings may be treated as errors in the
+    continuous integration system, which also runs those tests.
```

Comment:
```suggestion
    Tests are compiled automatically if you use the ``dev=yes`` SCons option.
    ``dev=yes`` is recommended if you plan on contributing to the engine
    development as it will automatically treat compilation warnings as errors.
    The continuous integration system will fail if any compilation warnings are detected,
    so you should strive to fix all warnings before opening a pull request.
```